vmm/src/lib.rs

```diff
@@ -644,6 +644,45 @@ impl VmmVersionInfo {
     }
 }
 
+#[derive(Debug, Clone)]
+struct MigrationState {
+    current_dirty_pages: u64,
+    downtime: Duration,
+    downtime_start: Instant,
+    iteration: u64,
+    iteration_cost_time: Duration,
+    iteration_start_time: Instant,
+    mb_per_sec: f64,
+    pages_per_second: u64,
+    pending_size: u64,
+    start_time: Instant,
+    threshold_size: u64,
+    total_time: Duration,
+    total_transferred_bytes: u64,
+    total_transferred_dirty_pages: u64,
+}
```

Comment:
I understand such struct is mostly for providing statistics with live-migration, and this is good to have. However, I don't see we need it for down-time limit support, say most of the members are internal to `memory_copy_iterations ` and never used anywhere else.

To keep the change small and self-contained, I'd drop such struct along with related changes on providing statistics.